### PR TITLE
Support newer httpx versions

### PR DESCRIPTION
**Pull request by nozik**
# Description

Support the latest versions of `httpx`, currently only versions lower than `0.19.0` are supported.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Adapting the existing tests of the previous versions of `httpx`.

# Does This PR Require a Core Repo Change?

- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added
- [x] Documentation has been updated

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-httpx/setup.cfg [hunk 1]
```diff
@@ -48,7 +48,6 @@ install_requires =
 test =
     opentelemetry-sdk ~= 1.3
     opentelemetry-test-utils == 0.27b0
-    respx ~= 0.17.0
```
Comment (ocelotl) [inline instrumentation/opentelemetry-instrumentation-httpx/setup.cfg:51]: Just curious, why is this dependency removed?

**Review by ocelotl — CHANGES_REQUESTED** [review]

**Comment by lzchen** [comment]
@jomasti 
Is it alright if you were to be added as a maintainer for this instrumentation? This was what I was adhering to in this [comment](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/461#issuecomment-853207677)

File: instrumentation/opentelemetry-instrumentation-httpx/setup.cfg (hunk 1 above)
Comment (nozik) [reply instrumentation/opentelemetry-instrumentation-httpx/setup.cfg:51]: The later versions of `httpx` are incompatible with `respx ~= 0.17.0`, so I added the different versions as dependencies`tox.ini`

**Comment by jomasti** [comment]
@lzchen I don't have the time to dedicate to any maintenance. It looks like this is being handled okay, and I wouldn't be providing anything extra.

File: instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py [hunk 2]
```diff
@@ -233,6 +233,36 @@ def _prepare_headers(headers: typing.Optional[Headers]) -> httpx.Headers:
...
+        headers = kwargs.get("headers", args[2] if len(args) > 2 else None)
+        stream = kwargs.get("stream", args[3] if len(args) > 3 else None)
+        extensions = kwargs.get(
+            "extensions", args[4] if len(args) > 4 else None
+        )
+
+    return method, url, headers, stream, extensions
+
+
+def inject_propagation_headers(headers, args, kwargs):
```
[the remark this answers is not in the document]
Comment (nozik) [reply instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py]: Done

File: instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py [hunk 3]
```diff
@@ -233,6 +233,36 @@ def _prepare_headers(headers: typing.Optional[Headers]) -> httpx.Headers:
     return httpx.Headers(headers)
 
 
+def _extract_parameters(args, kwargs):
```
[the remark this answers is not in the document]
Comment (nozik) [reply instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py:236]: Done

**Review by ocelotl — APPROVED** [review]

**Review by owais — APPROVED** [review]

**Comment by lonewolf3739** [comment]
@nozik to fix the docs issue please try adding an exception `httpx.Response` here https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/docs/nitpick-exceptions.ini#L22-L27.

**Comment by nozik** [comment]
@lonewolf3739 - that did the trick, thank you!
@ocelotl @owais PR is ready for merge. Thanks!

**Pull request merged**